editor/inserter/index.js

```diff
@@ -1,18 +1,31 @@
 function Inserter() {
 	const blocks = wp.blocks.getBlocks();
+	const blocksByCategory = blocks.reduce( ( groups, block ) => {
+		if ( ! groups[ block.category ] ) {
+			groups[ block.category ] = [];
+		}
+		groups[ block.category ].push( block );
+		return groups;
+	}, {} );
+	const categories = wp.blocks.getCategories();
 
 	return (
 		<div className="inserter">
 			<div className="inserter__arrow" />
 			<div className="inserter__content">
-				<div className="inserter__Category-blocks">
-					{ blocks.map( ( { slug, title, icon } ) => (
-						<div key={ slug } className="inserter__block">
-							<span className={ 'dashicons dashicons-' + icon } />
-							{ title }
+				{ categories
+					.map( ( category ) => !! blocksByCategory[ category.slug ] && (
+						<div key={ category.slug } className="inserter__Category-blocks">
```

Comment:
I assume the capitalization on "Category" `className` was unintentional. Maybe also we ought to avoid adding classes until we start styling them?